method/group.js

```diff
@@ -126,18 +126,44 @@ module.exports = function GroupMethods (ssb, keystore, state) {
       }]
 
       pull(
-        ssb.query.read({ query }),
+        ssb.query.read({ query: addedQuery }),
         pull.filter(addMemberSpec.isValid),
         pull.filter(msg =>
           groupId === msg.value.content.recps[0]
         ),
         pull.map(msg => msg.value.content.recps.slice(1)),
         pull.flatten(),
         pull.unique(),
-        pull.collect((err, members) => {
+        pull.collect((err, addedMembers) => {
           if (err) return cb(err)
 
-          return cb(null, members)
+          const excludedQuery = [{
+            $filter: {
+              value: {
+                content: {
+                  type: 'group/exclude-member'
+                }
+              }
+            }
+          }]
+
+          pull(
+            ssb.query.read({ query: excludedQuery }),
+            pull.filter(excludeMemberSpec.isValid),
+            pull.filter(msg =>
+              groupId === msg.value.content.recps[0]
+            ),
+            pull.map(msg => msg.value.content.excludes),
+            pull.flatten(),
+            pull.unique(),
+            pull.collect((err, excludedMembers) => {
+              if (err) return cb(err)
+
+              const members = addedMembers.filter(addedMember => !excludedMembers.includes(addedMember))
```

Comment:
This disallows people from being added back into a group. That's probably possible to allow, but we might have to fix the member tangle first. Make issue for later (there's already an issue for the tangle, link to that)

Reply:
Agree with this approach. I think you should leave a comment in the code like 
```suggestion
              const members = addedMembers.filter(addedMember => !excludedMembers.includes(addedMember))
              // NOTE: this currently prevents people who've been removed from being re-added
```